pyomo/core/plugins/transform/deactivate_trivial_constraints.py

```diff
@@ -0,0 +1,51 @@
+# -*- coding: UTF-8 -*-
+"""Transformation to deactivate trivial constraints."""
+from pyomo.core.base.constraint import Constraint
+from pyomo.core.kernel.numvalue import value
+from pyomo.core.plugins.transform.hierarchy import IsomorphicTransformation
+from pyomo.util.plugin import alias
+
+__author__ = "Qi Chen <qichen at andrew.cmu.edu>"
+
+
+class TrivialConstraintDeactivator(IsomorphicTransformation):
+    """Deactivates trivial constraints of form constant = constant."""
+
+    alias('core.deactivate_trivial_constraints', doc=__doc__)
+
+    def __init__(self):
+        """Initialize the transformation."""
+        super(TrivialConstraintDeactivator, self).__init__()
+        self._deactivated_constrs = set()
```

Comment:
In general I have been advocating that transformations should be stateless.  State needed for either undoing the transformation or backmapping should be stored on the model.